packages/remix-react/components.tsx

```diff
@@ -759,31 +764,143 @@ export type ScriptProps = Omit<
  */
 export function Scripts(props: ScriptProps) {
   let { manifest, serverHandoffString } = useRemixContext();
-  let { router } = useDataRouterContext();
+  let { router, static: isStatic } = useDataRouterContext();
   let { matches } = useDataRouterStateContext();
+  let staticContext = React.useContext(DataStaticRouterContext);
```

Comment:
Since we have to be defensive on `staticContext` anyway, should be just `const isStatic = staticContext != null`?